File: src/fiefapp.svelte/src/models/livingcondition.ts
interface ILivingcondition {
  id: string;
  type: LivingconditionType | undefined;
  name: string;
  baseCost: number;
  luxuryCost: number;
  focusGain: number;
  wellbeingGain: number;
  description: string;
}

export enum LivingconditionType {
  BARE = 'BARE',
  MEAGRE = 'MEAGRE',
  GOOD = 'GOOD',
  PLENTIFUL = 'PLENTIFUL',
  LUXURY = 'LUXURY',
}

export class Livingcondition implements ILivingcondition {
  id: string;
  type: LivingconditionType | undefined;
  name: string;
  baseCost: number;
  luxuryCost: number;
  focusGain: number;
  wellbeingGain: number;
  description: string;

  constructor(
    id?: string,
    type?: LivingconditionType,
    name?: string,
    baseCost?: number,
    luxuryCost?: number,
    focusGain?: number,
    wellbeingGain?: number,
    description?: string
  ) {
    this.id = id ? id : '';
    this.type = type ? type : undefined;
    this.name = name ? name : '';
    this.baseCost = baseCost ? baseCost : 0;
    this.luxuryCost = luxuryCost ? luxuryCost : 0;
    this.focusGain = focusGain ? focusGain : 0;
    this.wellbeingGain = wellbeingGain ? wellbeingGain : 0;
    this.description = description ? description : '';
  }
}

File: src/fiefapp.entities/Livingcondition.cs
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Livingcondition : IBaseEntity, IAlternativeEntity
    {
        public Guid Id { get; set; }
        public string Type { get; set; }
        public string Name { get; set; }
        public int BaseCost { get; set; }
        public int LuxuryCost { get; set; }
        public int FocusGain { get; set; }
        public int WellbeingGain { get; set; }
        public string Description { get; set; }
        public ICollection<Guid> GamesessionIds { get; set; }
    }
}
